### DISCUSSION TITLE
Broadcast - Display Where Received

### DISCUSSION THREAD

**Discussion by PianoMan7** [ScratchAddons/ScratchAddons#6797] [Ideas] [answered]
I would find it really amazing if there could be a way to right click a broadcast and see a list of all of the places where that message is being received. 

Then, from that list, if you click on one option, you are taken to that section of code in the editor.

The reverse could also be implemented for "when I receive" input blocks. You could see a list of places where the message is being sent from. 

As a comp sci educator, this would really speed up my workflow when I have to assess student work and I haven't sat down with them to see how each of them have built their code from start to finish!

**Comment by CST1229** [comment, marked as answer]
The "Editor find bar" addon has something like this, you can middleclick or shift+click a broadcast or receive block to see where that broadcast is used. Though it does not differentiate between `when I receive ()` and `broadcast ()` blocks.

**Reply by Sqrts321** [reply]
I didn't even know about that addon

**Comment by DNin01** [comment]
I feel like the developer tools addon or at least @griffpatch's Scratch 3 Developer Tools extension had a feature to highlight all sprites that contain a block referencing a message. I can't seem to find it Scratch Addons now. Have we ever had that feature?

**Reply by mxmou** [reply]
Looks like this was removed in v1.9.0 (PR #1390).
![image](https://github.com/ScratchAddons/ScratchAddons/assets/51849865/5b7be0e6-2724-4922-8810-6426bd0001cc)

**Reply by WorldLanguages** [reply]
I think griffpatch removed it because the find bar itself already has similar functionality. But multiple times I found this change to be less intuitive.

**Comment by griffpatch** [comment]
I don't think I ever added it. It was in the original scratch 2... But you
can just middle click a receiver to get the same functionality

**Reply by PianoMan7** [reply]
Okay that's great to know! I didn't know about either the shift+click method or the middle-click. Thank you!
